modules/assim.sequential/R/sda_plotting.R

```diff
@@ -845,3 +845,143 @@ post.analysis.multisite.ggplot <- function(settings, t, obs.times, obs.mean, obs
 }
 
 
+
+##' @rdname interactive.plotting.sda
+##' @export
+SDA_timeseries_plot <- function(ANALYSIS, FORECAST, obs.mean, obs.cov, outdir, pft.path = NULL, by = "site", CI = c(0.025, 0.975), 
+                                unit = list(AbvGrndWood = "Mg/ha", LAI = "m2/m2", SoilMoistFrac = "", TotSoilCarb = "kg/m2"),
+                                style = list(general_color = c("FORECAST" = "blue", "ANALYSIS" = "red", "OBS" = "black"),
+                                             fill_color = c("FORECAST" = "yellow", "ANALYSIS" = "green", "OBS" = "grey"),
+                                             title_color = "red")){
+  #TODO: make page, font, line, point sizes adjustable.
+  time_points <- names(FORECAST)
+  site_ids <- attributes(FORECAST[[1]])$Site
+  var_names <- attributes(FORECAST[[1]])$dimnames[[2]]
+  types <- c("FORECAST", "ANALYSIS", "OBS")
+  #new diag function: fixed the bug when length==1 then it will return 0x0 matrix
+  diag_fix <- function(vector){
+    if (length(vector)>1){
+      return(diag(vector))
+    }else if (length(vector)==1){
+      return(vector)
+    }
+  }
+  #read pft.csv file for the option by == pft.
+  if(!is.null(pft.path)){
+    pft <- read.csv(pft.path)
+  }
+  #create database
+  DB <- data.frame()
+  for (id in sort(unique(site_ids))) {
```

Comment:
not essential, but I suspect a `lapply` or `map` approach would be a lot quicker than this level of nested looping. I also bet that within a timepoint and site you could handle the variable reorganization with pivot_wider and/or pivot_longer